### DISCUSSION TITLE
Apache AJP and EJBCA modules question

### DISCUSSION THREAD

**Discussion by kpeirce** [Keyfactor/ejbca-ce#573] [General]
I am trying to get Apache2 AJP to proxy for EJBCA on Wildfly 26.  I am using Keyfactor documentation. I know Apache is OK because when I browsed to 127.0.0.1:443 I got the "Service Unavailable" page.  I subsequently added an AJP listener for WildFly using the instructions at https://undertow.io/undertow-docs/undertow-docs-2.0.0/listeners.html and github's Undertow_AJP_listeners.adoc. Now I get a 404 error.  Are there any other examples I could look at? Also, the OS is Ubuntu 22.04. I have a YubiHSM2. I believe that EJBCA should be able to run without the HSM connection working. Is that a valid approach to take?

**Comment by kpeirce** [comment]
Also, this is a standalone CA. I do have mod_proxy enabled. Do I need mod_jk and mod_cluster?

**Reply by primetomas** [reply]
It was a while since I configured an apache myself and things change. But don't you need mod_proxy_ajp? 
https://httpd.apache.org/docs/2.4/mod/mod_proxy_ajp.html

**Reply by kpeirce** [reply]
Hi I am using the Keyfactor/Primekey documentation that says I must have: mod_proxy, mod_proxy_http, mod_proxy_ajp, mod_proxy_balancer, mod_proxy_rewrite and mod_ssl.

**Reply by primetomas** [reply]
Can you link to this documentation?
